### DISCUSSION TITLE
Vector should have min(...) max(...) and abs()

### DISCUSSION THREAD

**Discussion by MichalLebeda** [hecrj/iced#1457] [Ideas]
We should be able to create a new vector taking a minimum/maximum from two input vectors (component wise).
```
min((0,2), (1,0)) => (0, 0)
max((0,2), (1,0)) => (1, 2)
```

Calculating absolute vector should also be possible.
```
abs((-2, 1)) => (2,1)
```

**Comment by 13r0ck** [comment]
This doesn't seem related to iced? Is this an issue you are having with rust lang itself?

**Comment by bungoboingo** [comment]
I think he is referring to `iced_core::Vector`. If that is the case, implementing this would be pretty quick. In the meantime you can just wrap `iced_core::Vector` in a struct and impl min/max/abs yourself.

**Reply by MichalLebeda** [reply]
Yes I do, sorry for not being clear. I may implement it tommorow if but not sure how to make it universal as Vector can use different primitive types right? Also, does min/max definition stated above make sense to you? I wonder whether it may be confused with returning min/max of x and y of single vector.
